src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/entity/ActionPlan.kt

```diff
@@ -32,6 +33,10 @@ data class ActionPlan(
   @ManyToOne @Fetch(FetchMode.JOIN) var submittedBy: AuthUser? = null,
   var submittedAt: OffsetDateTime? = null,
 
+  // This isn't set directly and hence it's a 'val'. Once an action plan appointment has been persisted, on retrieval
+  // of the associated action plan this list will be automatically set by hibernate.
+  @OneToMany(mappedBy = "actionPlan") val appointments: List<ActionPlanAppointment>,
```

Comment:
i read somewhere that using `Set` results in more efficient SQL than `List` https://vladmihalcea.com/hibernate-facts-favoring-sets-vs-bags/

Reply:
On reflection a bi-directional relationship is not necessary. At present, an action plan doesn't need to know about its appointments as it's not part of the action plan dto returned to the ui. Appointments can be retrieved directly given the requirements of the current set of endpoints. So I have removed the list (which should have been a set) for now.